src/Symfony/Component/Validator/ValidatorBuilder.php

```diff
@@ -217,27 +217,14 @@ public function addMethodMappings(array $methodNames)
     /**
      * Enables annotation based constraint mapping.
      *
-     * @param bool $skipDoctrineAnnotations
-     *
      * @return $this
      */
-    public function enableAnnotationMapping(/* bool $skipDoctrineAnnotations = true */)
+    public function enableAnnotationMapping()
     {
         if (null !== $this->metadataFactory) {
             throw new ValidatorException('You cannot enable annotation mapping after setting a custom metadata factory. Configure your metadata factory instead.');
         }
 
-        $skipDoctrineAnnotations = 1 > \func_num_args() ? false : func_get_arg(0);
-        if (false === $skipDoctrineAnnotations || null === $skipDoctrineAnnotations) {
-            trigger_deprecation('symfony/validator', '5.2', 'Not passing true as first argument to "%s" is deprecated. Pass true and call "addDefaultDoctrineAnnotationReader()" if you want to enable annotation mapping with Doctrine Annotations.', __METHOD__);
```

Comment:
I don't get why passing `true` as first argument is needed here?

Reply:
/cc @derrabus (from "git blame")

Reply:
Well, at first, we had a `enableAnnotationMapping(Reader reader = null)` signature to enable annotation mapping either with a custom reader or a default doctrine reader.
https://github.com/symfony/symfony/commit/441c80603e77ceccb1e56db80086220fd80b92ca made it possible to enable the annotation mapping without a doctrine reader at all, as our annotation mapping is responsible for both attributes and annotations (if you don't pass a reader, it will support only attributes). The `true` argument was there to allow the new way without breaking BC (as calling without argument or with null was already a thing)

Reply:
@stof Thank you for the explanation :-)